### DISCUSSION TITLE
Compute rect values from stream

### DISCUSSION THREAD

**Discussion by meghanaviyyapu** [pymupdf/PyMuPDF#2330] [Looking for help]
Can you please let me know how rect values can be computed from the below stream values?

 EMC  /P <</MCID 9>> BDC 0.969 0.588 0.275 rg
87.75 428.79 m
151.88 428.79 l
151.88 431.98 l
158.25 425.6 l
151.88 419.23 l
151.88 422.42 l
87.75 422.42 l
h
f*
2 w
0.714 0.427 0.192 RG

87.75 428.79 m
151.88 428.79 l
151.88 431.98 l
158.25 425.6 l
151.88 419.23 l
151.88 422.42 l
87.75 422.42 l
h
S
 EMC

**Comment by JorjMcKie** [comment]
The `m` ("move") command positions at the point with the preceding 2 float coordinates. The following `l` ("line") commands draw a line up to the point given by the preceding 2 floats, which becomes the current point afterwards. Etc.
So the resulting rectangle is the union of all these points up to command `h`.

Please note that all coordinates are probably given in PDF standard geometry: point (0,0) is bottom-left. To compute MuPDF coordinates, you need the page height and replace every y-coordinate by `height - y`.
